contracts/MarketMakerProxy.sol

```diff
@@ -4,102 +4,89 @@ pragma solidity 0.7.6;
-        _;
-    }
+    address public signer;
+    mapping(address => bool) public isWithdrawWhitelist;
 
-    constructor() Ownable(msg.sender) {
-        operator = msg.sender;
+    constructor(
+        address _owner,
+        address _signer,
+        IWETH _weth
+    ) Ownable(_owner) {
+        require(_signer != address(0), "MarketMakerProxy: zero address");
+        signer = _signer;
+        WETH = _weth;
     }
 
     receive() external payable {}
 
-    // Manage
-    function setSigner(address _signer) public onlyOperator {
-        SIGNER = _signer;
+    function setSigner(address _signer) external onlyOwner {
+        require(_signer != address(0), "MarketMakerProxy: zero address");
+        emit ChangeSigner(_signer);
+        signer = _signer;
     }
 
-    function setConfig(address _weth) public onlyOperator {
-        WETH_ADDR = _weth;
-    }
-
-    function setWithdrawer(address _withdrawer) public onlyOperator {
-        withdrawer = _withdrawer;
-    }
-
-    function setOperator(address _newOperator) public onlyOwner {
-        operator = _newOperator;
+    function setAllowance(address[] calldata tokenAddrs, address spender) external onlyOwner {
+        for (uint256 i = 0; i < tokenAddrs.length; i++) {
+            IERC20(tokenAddrs[i]).safeApprove(spender, LibConstant.MAX_UINT);
+        }
     }
 
-    function setAllowance(address[] memory token_addrs, address spender) public onlyOperator {
-        for (uint256 i = 0; i < token_addrs.length; i++) {
-            address token = token_addrs[i];
-            IERC20(token).safeApprove(spender, LibConstant.MAX_UINT);
+    function closeAllowance(address[] calldata tokenAddrs, address spender) external onlyOwner {
+        for (uint256 i = 0; i < tokenAddrs.length; i++) {
+            IERC20(tokenAddrs[i]).safeApprove(spender, 0);
         }
     }
 
-    function closeAllowance(address[] memory token_addrs, address spender) public onlyOperator {
-        for (uint256 i = 0; i < token_addrs.length; i++) {
-            address token = token_addrs[i];
-            IERC20(token).safeApprove(spender, 0);
-        }
+    function updateWithdrawWhitelist(address _addr, bool _enabled) external onlyOwner {
+        require(_addr != address(0), "MarketMakerProxy: zero address");
+        isWithdrawWhitelist[_addr] = _enabled;
+        emit UpdateWhitelist(_addr, _enabled);
     }
 
-    function registerWithdrawWhitelist(address _addr, bool _add) public onlyOperator {
-        isWithdrawWhitelist[_addr] = _add;
+    function wrapETH() external onlyOwner {
+        uint256 balance = address(this).balance;
+        if (balance > 0) {
+            WETH.deposit{ value: balance }();
+            emit WrapETH(balance);
+        }
     }
 
-    function withdraw(
+    function withdrawToken(
         address token,
-        address payable to,
+        address to,
         uint256 amount
-    ) public onlyWithdrawer {
+    ) external onlyOwner {
```

Comment:
I think this changes the usage pattern of MMP? If this is not customized on a per marker maker basis, maybe we need to think about if this suits their need?

Owner in my opinion would be a cold wallet/multi-sig, requiring it to execute a transaction each time to withdraw one asset might be a problem.

Reply:
Right, I didn't know this consideration though. However if the `withdrawer` and `operator` are both EOA, what's the purpose to have a cold wallet or multisig owner since the allowance and the withdraw whitelist security is actually relied on EOA accounts?

Reply:
Yeah, indeed the access control seems too simple to be effectively secure. If owner need to set withdrawer and also need to withdraw himself, then it doesn't seem like there's a need to set the withdrawer at all.🤔 

But if withdrawer can just withdraw all funds, then this extra layer of access control not help with the security at all, just like you point out.

I think there's two way forward, either we add in more fine-grain access control so adding withdrawers do help improve security or we just remove withdrawer completely. I prefer the latter.😂 MM can add in more fine-grain access control if they want.

Reply:
Yeah actually I think this MMP should be just an example. The access control design is heavily depends on the need of each MM. I think the withdraw whitelist is still make sense even without withdrawer?

Reply:
Unless withdraw is called by non owner? Otherwise owner can just specify any `to` he likes since he is also the one controls whitelist?

Reply:
Yeah since the owner controls everything. The whitelist here is kind of two step process to avoid typo or incorrect copy and paste. It's like the experience of CEX. Perhaps we can add a time lock in `updateWithdrawWhitelist()`?

Reply:
I think we can just leave it to MMs.